src/operation/count/mod.rs

```diff
@@ -52,8 +75,29 @@ impl Operation for Count {
         ))
     }
 
-    fn handle_response(&self, response: CommandResponse) -> Result<Self::O> {
-        response.body::<ResponseBody>().map(|body| body.n)
+    fn handle_response(
+        &self,
+        response: CommandResponse,
+        description: &StreamDescription,
+    ) -> Result<Self::O> {
+        let response_body: ResponseBody = match description.max_wire_version {
+            Some(v) if v >= SERVER_4_9_0_WIRE_VERSION => {
+                let CursorBody { mut cursor } = response.body()?;
+
+                cursor
+                    .first_batch
+                    .pop_front()
+                    .and_then(|doc| bson::from_document(doc).ok())
+                    .ok_or_else(|| {
+                        Error::from(ErrorKind::ArgumentError {
```

Comment:
I think for this case we typically use `ResponseError`, though honestly we could just replace any usages of `ResponseError` with `InternalError` if we wanted to.

Reply:
Oops, good catch